lib/govuk_publishing_components/presenters/breadcrumb_selector.rb

```diff
@@ -0,0 +1,105 @@
+module GovukPublishingComponents
+  module Presenters
+    class BreadcrumbSelector
+      def self.call(*args)
+        new(*args).output
+      end
+
```

Comment:
A question rather than a comment: I would have made everything in this presenter private, except the `call` method. Is there no point doing that in this case?

Reply:
I tend to use `private` fairly sparingly. It's not particular meaningful in Ruby as it's very easy to bypass (`send(:private_method_name)`). I also think the greatest complement is for someone to reuse code I've written in a way I hadn't imagined. So, I tend to assume that as I don't know how someone else might want to use the object I've built, then why should I guess which methods might or might not be useful to them. Also leaving them public makes debugging easier.

Therefore, I tend only to use `private` where using the method would break things or has security implication. For example, in controllers for methods that aren't actions and so could be exposed as actions due to a routing error if they weren't private.

Reply:
Thanks